### PR TITLE
finish test

### PR DESCRIPTION
**Pull request by rmanela**

### CODE REVIEW

File: modules/ecs6-class/line.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ const Point = require("./point");
 
 class Line {
     constructor({ point1 = new Point(), point2 = new Point(), n = undefined, slope = undefined }) {
+        console.log(typeof(point1), typeof(point2), "typeooooooooooop1p2p2p1p");
+
+        // if (typeof (x) != "number" || typeof (y) != "number") {
+        //     throw new Error("aaaa")
+        // }
```
Comment (gemtechd) [inline modules/ecs6-class/line.js]: don't push console and no remarks

File: modules/ecs6-class/line.js [hunk 2]
```diff
@@ -2,6 +2,11 @@ const Point = require("./point");
 
 class Line {
     constructor({ point1 = new Point(), point2 = new Point(), n = undefined, slope = undefined }) {
```
Comment (gemtechd) [inline modules/ecs6-class/line.js:4]: What happens when point1 is a string when n is a string ?? think about more pitfalls

File: modules/ecs6-class/line.js [hunk 3]
```diff
@@ -13,10 +18,18 @@ class Line {
     }
 
     calculateNOfLineFunction = () => {
+
+        if (this.slope === undefined) {
+            throw new Error("the slope must be defined")
```
Comment (gemtechd) [inline modules/ecs6-class/line.js:20]: you can calculate it

File: modules/ecs6-class/line.js [hunk 4]
```diff
@@ -13,10 +18,18 @@ class Line {
...
+        }
+        if (typeof (this.slope) != "number") {
+            throw new Error("the slope must be number")
```
Comment (gemtechd) [inline modules/ecs6-class/line.js:23]: write !==

File: modules/ecs6-class/line.js [hunk 5]
```diff
@@ -26,13 +39,26 @@ class Line {
 
 
     getPointByX(x) {
+        if (x === undefined) {
+            throw new Error("the function getPointByX must get x")
+        }
+        if (typeof (x) != "number") {
+            throw new Error("the function getPointByX must get x of type number")
+        }
         let y = this.slope * x + this.n
```
Comment (gemtechd) [inline modules/ecs6-class/line.js:57]: what happens if the slope is not defined?

File: modules/ecs6-class/line.js [hunk 6]
```diff
@@ -26,13 +39,26 @@ class Line {
...
-        return new Point({ x, y })
+        return new Point({ x: x, y: y })
```
Comment (gemtechd) [inline modules/ecs6-class/line.js:58]: what is the different between the `{x, y}` and `{x:x, y:y}`

File: modules/ecs6-class/line.js [hunk 7]
```diff
@@ -26,13 +39,26 @@ class Line {
...
 
     getPointByY(y) {
+        if (y === undefined) {
+            throw new Error("the function getPointByY must get y")
+        }
+        if (typeof (y) != "number") {
+            throw new Error("the function getPointByY must get y of type number")
+        }
+
         let x = (y - this.n) / this.slope;
```
Comment (gemtechd) [inline modules/ecs6-class/line.js:69]: check if slope and n are defined

File: modules/ecs6-class/point.js [hunk 8]
```diff
@@ -1,14 +1,33 @@
 class Point {
-    constructor({x=0, y=0}={}) {
+    constructor({ x = 0, y = 0 } = {}) {
+        if (typeof (x) != "number" || typeof (y) != "number") {
+            throw new Error("the argument must be type number")
```
Comment (gemtechd) [inline modules/ecs6-class/point.js:4]: Which one is not a number?

File: modules/geometry-calculation.js [hunk 9]
```diff
@@ -2,12 +2,15 @@ const Line = require('./ecs6-class/line')
 
 const calculateDistance = (point1, point2) => {
     let distanceX = (point2.x - point1.x) ** 2;
```
Comment (gemtechd) [inline modules/geometry-calculation.js:11]: check the arguments

File: modules/geometry-calculation.js [hunk 10]
```diff
@@ -2,12 +2,15 @@ const Line = require('./ecs6-class/line')
...
-    let distanceY = (point2.y - point2.y) ** 2;
+    let distanceY = (point2.y - point1.y) ** 2;
     const distance = Math.sqrt(distanceX + distanceY);
     return distance;
 }
 
 const calculateJunctionPoint = (line1, line2) => {
+
```
Comment (gemtechd) [inline modules/geometry-calculation.js]: check the arguments

File: modules/geometry-calculation.js [hunk 11]
```diff
@@ -2,12 +2,15 @@ const Line = require('./ecs6-class/line')
...
     return distance;
 }
 
 const calculateJunctionPoint = (line1, line2) => {
+    
+
+
     if (line1.slope === line2.slope) {
         if (line1.n === line2.n) {
             return true
```
Comment (gemtechd) [inline modules/geometry-calculation.js:27]: check slope and n

File: modules/geometry-calculation.js [hunk 12]
```diff
@@ -24,17 +27,20 @@ const calculateJunctionPoint = (line1, line2) => {
 }
 
 const isPointOnLine = (line, point) => {
```
Comment (gemtechd) [inline modules/geometry-calculation.js:40]: check the arguments

File: modules/geometry-calculation.js [hunk 13]
```diff
@@ -24,17 +27,20 @@ const calculateJunctionPoint = (line1, line2) => {
...
-    const proxyLine = new Line({ point1: line.point1, point2: point })
+
+    const proxyLine = new Line({ point1: line.point1, point2: point, n: undefined, slope: undefined })
     proxyLine.calculateSlope()
     if (line.slope === proxyLine.slope) {
         proxyLine.calculateNOfLineFunction()
```
Comment (gemtechd) [inline modules/geometry-calculation.js:45]: check the slope and n of line

File: package.json [hunk 14]
```diff
@@ -4,7 +4,8 @@
   "description": "practice unit tests in javascript",
   "main": "index.js",
   "scripts": {
-    "test": "jest"
+    "test": "jest",
+    "test:coverage":"jest --coverage"
   },
   "dependencies": {
     "jest": "^29.7.0"
```
Comment (gemtechd) [inline package.json:14]: jest does not belong to the dependencies

File: test/modules/ecs6-class/line.test.js [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+            expect(l1.point2.y).toEqual(4)
+            expect(l1.n).toEqual(2)
+            expect(l1.slope).toEqual(-3)
+            expect(l2.point1.x).toEqual(5)
+            expect(l2.point1.y).toEqual(5)
+            expect(l2.point2.x).toEqual(6)
+            expect(l2.point2.y).toEqual(6)
+            expect(l2.n).toBe(undefined)
+            expect(l2.slope).toBe(undefined)
+
```
Comment (gemtechd) [inline test/modules/ecs6-class/line.test.js]: don't write so many tests in one time
divide them in a few blocks

File: test/modules/ecs6-class/line.test.js [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+            let point1 = new Point({ x: 5, y: 5 })
+            let point2 = new Point({ x: 6, y: 6 })
+            let l1 = new Line({ point1: p1, point2: p2, n: 2, slope: -3 })
+            let l2 = new Line({ point1: point1, point2: point2, })
+            expect(l1.slope).toEqual(-3)
+
+            l1.calculateSlope()
+            l2.calculateSlope()
+            expect(l1.slope).toEqual(1)
+            expect(l2.slope).toEqual(1)
```
Comment (gemtechd) [inline test/modules/ecs6-class/line.test.js]: when you test two cases, separate it into two tests

File: test/modules/ecs6-class/line.test.js [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+
+
+    describe("GET_POINT_ON_Y_ASIS", function () {
+        it("should retutn a new point on y", function () {
+
+
+            let line1 = new Line({ point1: new Point({ x: 1, y: 6 }), point2: new Point({ x: 2, y: 3 }), })
+            line1.calculateSlope()
+            line1.calculateNOfLineFunction()
+            expect(line1.getPointOnYAsis()).toStrictEqual(new Point({ x: 0, y: 9 }))
```
Comment (gemtechd) [inline test/modules/ecs6-class/line.test.js:77]: if you test a certain function, don't call other ones

File: test/modules/ecs6-class/line.test.js [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+        })
+    })
+
+    describe("GET_POINT_ON_X_ASIS", function () {
+        it("should retutn a new point on x", function () {
+
+            let line1 = new Line({ point1: new Point({ x: 1, y: 6 }), point2: new Point({ x: 2, y: 3 }), })
+            line1.calculateSlope()
+            line1.calculateNOfLineFunction()
+            expect(line1.getPointOnXAsis()).toStrictEqual(new Point({ x: 3, y: 0 }))
```
Comment (gemtechd) [inline test/modules/ecs6-class/line.test.js:84]: when you test a certain function, don't call other functions

File: test/modules/ecs6-class/line.test.js [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+    describe("ERROR", function () {
+        let line = new Line({ point1: new Point({ x: 3, y: 7 }), point2: new Point({ x: 9, y: 7 }), n: 9 })
+        let line2 = new Line({ point1: new Point({ x: 3, y: 7 }), point2: new Point({ x: 9, y: 7 }), n: 9, slope: "jskjk" })
+        let line3 = new Line({ point1: new Point({ x: 3, y: 7 }), point2: new Point({ x: 9, y: 7 }), n: 9, slope: 6 })
+        let line4 = new Line({ point1: "huhj", point2: "jij", n: "9", slope: "ijdi" })
+        console.log(typeof (line4.point1));
+        // it("the constractor of line must get 2 points of type number",function(){
+        //     expect(()=>{new Line({ point1: new Point({ x: 3, y: 7 }), point2: new Point({ x: 9, y: 7 }), n: 9 ,slope:"ui"})}).toThrow("aaaa")
+        // })
+
```
Comment (gemtechd) [inline test/modules/ecs6-class/line.test.js]: don't push remarks and consoles

File: test/modules/ecs6-class/line.test.js
Comment (gemtechd) [inline test/modules/ecs6-class/line.test.js:1]: missing test errors

File: test/modules/geomatry-calculation.test.js
Comment (gemtechd) [inline test/modules/geomatry-calculation.test.js:1]: 1. Where are the mocks???

**Review by gemtechd — COMMENTED** [review]
See the comments
see in the tests to separate test cases 
don't execute functions that are not tested (that's why we test one function, you have to see that the function will execute inside it everything needed)

File: test/modules/ecs6-class/line.test.js (hunk 16 above)
Comment (rmanela) [reply test/modules/ecs6-class/line.test.js]: I don't understand why to divide into 2 cases

Comment (gemtechd) [reply test/modules/ecs6-class/line.test.js]: It's a way of writing code

**Comment by rmanela** [comment]
i want get more task

**Pull request closed without merging**